awx/main/models/events.py

```diff
@@ -538,23 +539,38 @@ def _update_host_summary_from_stats(self, hostnames):
 
             from awx.main.models import Host, JobHostSummary, HostMetric  # circular import
 
-            all_hosts = Host.objects.filter(pk__in=self.host_map.values()).only('id', 'name')
+            if self.job.inventory.kind == 'constructed':
+                all_hosts = Host.objects.filter(id__in=self.job.inventory.hosts.values_list(Cast('instance_id', output_field=models.IntegerField()))).only(
+                    'id', 'name'
+                )
+                constructed_host_map = self.host_map
+                host_map = {}
+                for host in all_hosts:
+                    host_map[host.name] = host.id
+            else:
+                all_hosts = Host.objects.filter(pk__in=self.host_map.values()).only('id', 'name')
+                constructed_host_map = {}
+                host_map = self.host_map
+
             existing_host_ids = set(h.id for h in all_hosts)
 
             summaries = dict()
             updated_hosts_list = list()
             for host in hostnames:
                 updated_hosts_list.append(host.lower())
-                host_id = self.host_map.get(host, None)
+                host_id = host_map.get(host, None)
```

Comment:
nit-pick and not new, but `None` is the default, so:

```suggestion
                host_id = host_map.get(host)
```